imap_processing/io.py

```diff
@@ -0,0 +1,162 @@
+"""Input/output capabilities for the IMAP data processing pipeline."""
```

Comment:
I was thinking about holding off on adding those descriptions until we decided on using this as a script versus just functions, because right now it can't actually be used as a script. Thoughts on whether we want to add a script or other info here now versus later?